cmd/zc_traverser_file.go

```diff
@@ -101,24 +101,22 @@ func (t *fileTraverser) traverse(processor objectProcessor, filters []objectFilt
 			for _, fileInfo := range lResp.FileItems {
 				f := currentDirURL.NewFileURL(fileInfo.Name)
 
-				//// TODO: the cost is high while otherwise we cannot get the last modified time. As Azure file's PM description, list might get more valuable file properties later, optimize the logic after the change...
-				//// TODO this traverser is only being used by rm at the moment, so we don't need the properties, uncomment in the future when this is no longer true
-				//fileProperties, err := f.GetProperties(t.ctx)
-				//if err != nil {
-				//	return err
-				//}
+				fileProperties, err := f.GetProperties(t.ctx)
+				if err != nil {
+					return err
+				}
 
 				// compute the relative path of the file with respect to the target directory
 				fileURLParts := azfile.NewFileURLParts(f.URL())
 				relativePath := strings.TrimPrefix(fileURLParts.DirectoryOrFilePath, targetURLParts.DirectoryOrFilePath)
 				relativePath = strings.TrimPrefix(relativePath, common.AZCOPY_PATH_SEPARATOR_STRING)
 
 				storedObject := storedObject{
-					name:         getObjectNameOnly(fileInfo.Name),
-					relativePath: relativePath,
-					//lastModifiedTime: fileProperties.LastModified(),
-					//md5:              fileProperties.ContentMD5(),
-					//size:             fileProperties.ContentLength(),
+					name:             getObjectNameOnly(fileInfo.Name),
```

Comment:
Changing to the newStoredObject func, I will not revert the getProperties call due to
- Potential future usage in sync
- Getting these properties in STE feels somewhat janky when all other cases get these properties in the enumerator
- The only reason we didn't get properties here is because it was only `rm` that was using it.